File: FMA/FMA-UI/src/app/models/chauffeur.ts
import { ChauffeurTankkaart} from "./chauffeur-tankkaart";
import { Adres } from "./adres";
import { Geslacht } from "./enums/geslacht";
import { RijbewijsType } from "./enums/rijbewijsType";
import { GemeldeSchade } from "./gemelde-schade";
import { Gemeente } from "./gemeente";

export interface Chauffeur {
    id: string;
    naam: string;
    voornaam: string;
    email: string;
    geboortedatum: string;
    rijksregisternummer: string;
    isActief: boolean;
    tankkaart: ChauffeurTankkaart | null;
    adres: Adres;
    gemeente: Gemeente | null;
    geslacht: Geslacht;
    rijbewijsType: RijbewijsType | null;
    gemeldeSchades: GemeldeSchade[];
}

File: FMA/FMA.Domain/Entities/Chauffeur.cs
using FMA.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FMA.Domain.Entities
{
    public class Chauffeur : BaseEntity
    {
        public string Naam { get; set; } = string.Empty;
        public string Voornaam { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public DateTime Geboortedatum { get; set; }
        public string Rijksregisternummer { get; set; } = string.Empty;
        public bool IsActief { get; set; } = true;
        public RijbewijsType? RijbewijsType { get; set; }
        
        public Geslacht Geslacht { get; set; }

        public virtual Tankkaart? Tankkaart { get; set; }
        public virtual Adres? Adres { get; set; }
        public virtual Gemeente? Gemeente { get; init; }
        public virtual List<Voertuig> Voertuigen { get; set; }
        public virtual List<Aanvraag> Aanvragen { get; init; }
        public virtual List<Inspectieverslag> Inspectieverslagen { get; init; }
        public virtual List<GemeldeSchade> GemeldeSchades { get; init; }

    }
}
